test/models/panel_survey_test.exs

```diff
@@ -0,0 +1,244 @@
+
+    assert result == :ok
+    assert panel_survey.name == name
+    assert panel_survey.project_id == project.id
+    panel_survey = Repo.preload(panel_survey, :project)
+    assert panel_survey.project == project
+  end
+
+
+  test "create_panel_survey/1 with valid data creates a panel_survey inside a folder" do
+    project = insert(:project)
+    folder = insert(:folder, project: project)
+    name = @foo_string
+
+    {result, panel_survey} =
+      PanelSurvey.create_panel_survey(%{name: name, project_id: project.id, folder_id: folder.id})
+
+    assert result == :ok
+    assert panel_survey.folder_id == folder.id
+    panel_survey = Repo.preload(panel_survey, [folder: :project])
+    assert panel_survey.folder == folder
+  end
+
+  test "list_panel_surveys/0 returns all panel_surveys" do
+    panel_survey = dummy_panel_survey()
+
+    listed_panel_surveys = PanelSurvey.list_panel_surveys()
+
+    assert listed_panel_surveys == [panel_survey]
+  end
+
+  test "get_panel_survey!/1 returns the panel_survey with given id" do
+    panel_survey = dummy_panel_survey()
+
+    loaded_panel_survey = PanelSurvey.get_panel_survey!(panel_survey.id)
+
+    assert loaded_panel_survey == panel_survey
+  end
+
+  test "create_panel_survey/1 with invalid data returns error changeset (no name)" do
+    project = insert(:project)
+    invalid_attrs = %{project_id: project.id}
+
+    {result, changeset} = PanelSurvey.create_panel_survey(invalid_attrs)
+
+    assert result == :error
+    assert changeset.action == :insert
+    assert changeset.changes == %{project_id: project.id}
+    assert changeset.valid? == false
+    assert changeset.errors == [name: {"can't be blank", [validation: :required]}]
+  end
+
+  test "create_panel_survey/1 with invalid data returns error changeset (no project)" do
+    name = @foo_string
+    invalid_attrs = %{name: name}
+
+    {result, changeset} = PanelSurvey.create_panel_survey(invalid_attrs)
+
+    assert result == :error
+    assert changeset.action == :insert
+    assert changeset.changes == %{name: "foo"}
+    assert changeset.valid? == false
+    assert changeset.errors == [project_id: {"can't be blank", [validation: :required]}]
+  end
+
+  test "update_panel_survey/2 with valid data updates the panel_survey (name)" do
+    panel_survey = dummy_panel_survey()
+    updated_name = @bar_string
+    update_attrs = %{name: updated_name}
+
+    {result, updated_panel_survey} = PanelSurvey.update_panel_survey(panel_survey, update_attrs)
+
+    # The right panel survey was updated
+    assert result == :ok
+    assert panel_survey.id == updated_panel_survey.id
+
+    # The name was updated
+    refute panel_survey.name == updated_panel_survey.name
+    assert updated_panel_survey.name == updated_name
+  end
+
+  # TODO: updating a panel survey project shouldn't be allowed.
+  # Where do we put this restriction. At the model level or at the controller level?
```

Comment:
The business logic should not be in the controller, thus, the restriction should be in the model or action

Reply:
do you have this restriction in mind for the new changes?

Reply:
> the restriction should be in the model
Done!